docs/design/lexical_conventions/symbolic_tokens.md

```diff
@@ -0,0 +1,185 @@
+| `->`  | Return type and `p->x` equivalent to `(*p).x` (in C++)                                                     |
+| `=>`  | Match syntax                                                                                               |
+| `[]`  | Subscript                                                                                                  |
+| `()`  | Function call and function declaration                                                                     |
+| `{}`  | Struct literals, blocks of control flow statements and the bodies of definitions (classes, functions, etc) |
+| `,`   | Separate arguments in a function call, elements of a tuple, or parameters of a function declaration        |
+| `.`   | Member access                                                                                              |
+| `:`   | Scope                                                                                                      |
+
+This list is expected to grow over time as more symbolic tokens are required by
+language proposals.
+
+Note: The above list only covers up to
+[#601](https://github.com/carbon-language/carbon-lang/pull/601) and more have
+been added since that are not reflected here.
+
+### Whitespace
+
+to support the use of the same symbolic token as a prefix operator, an infix
+operator, and a postfix operator (in some cases) we want a rule that allows us
+to simply and unambiguously parse operators that might have all three fixities.
+
+For example, given the expression `a * - b`, there are two possible parses:
+
+-   As `a * (- b)`, multiplying `a` by the negation of `b`.
+-   As `(a *) - b`, subtracting `b` from the pointer type `a *`.
+
+The chosen rule to distinguish such cases is to consider the presence or absence
+of whitespace, as it strikes a good balance between simplicity and
+expressiveness for the programmer and simplicity and good support for error
+recovery in the implementation. Hence `a * -b` uses the first interpretation,
+`a* - b` uses the second interpretation, and other combinations (`a*-b`,
+`a *- b`, `a* -b`, `a * - b`, `a*- b`, `a *-b`) are rejected as errors.
+
+We require whitespace to be present or absent around the operator to indicate
+its fixity, as this is a cue that a human reader would use to understand the
+code: binary operators have whitespace on both sides, and unary operators lack
+whitespace between the operator and its operand.
+
+But in some cases omitting the whitespace around a binary operator aids
+readability, such as in expressions like `2*x*x + 3*x + 1`, hence we have an
+allowance in such cases. In this case the operator with whitespace on neither
+side, if the token immediately before the operator indicates it is the end of an
+operand, and the token immediately after the operator indicates it is the
+beginning of an operand, the operator is treated as binary.
```

Comment:
```suggestion
While it is generally recommended to use whitespace to improve readability, there are cases where omitting whitespace can actually aid in readability. One such case is when you have a long expression with nested parentheses and operators.
`(2*x+1)/((x+1)*(x-1))-3*(x-2)`

While this expression is valid and follows the rules of the Carbon syntax, it can be difficult to read due to the nested parentheses and operators. However, if we omit whitespace around the binary operators, we can make the expression more readable:
`(2*x + 1) / ((x + 1) * (x - 1)) - 3 * (x - 2)`

In this version of the expression, the lack of whitespace around the operators actually makes it easier to see the grouping of terms, and the overall structure of the expression.
```

Reply:
I disagree -- this is not a language reference but a design document, and rationale is definitely part of that.

Reply:
Yes, that's why I included "and why", but by my reading @satakshigarg's suggestion is not giving a rationale, it's giving advice about how best to use this feature.

Reply:
Just to be clear, I'm saying that I don't think any change is needed here.